Broken on MacOS 13

ref https://github.com/nim-lang/Nim/actions/runs/12685916483/job/35357196370
```
  nimble make_lib
    Executing task make_lib in /Users/runner/work/Nim/Nim/pkgstemp/constantine/constantine.nimble
  Compiling static library:  lib/libconstantine.arm64.a
    compiler config: --cpu:arm64 -l:'-target arm64-apple-macos11' -t:'-target arm64-apple-macos11' --tlsEmulation=off  -d:danger  --panics:on -d:noSignalHandler  --mm:arc -d:useMalloc  --verbosity:0 --hints:off --warnings:off  --passC:-fno-semantic-interposition  --passC:-falign-functions=64  --passC:-fmerge-all-constants
  /Users/runner/work/Nim/Nim/pkgstemp/constantine/nimcache/libconstantine_static/@m..@sconstantine@smath@sarithmetic@sassembly@slimbs_asm_bigint_arm64.nim.c:49:51: error: invalid output constraint '=@cccc' in asm
  : [v1]"=&r" (v1), [u1]"=&r" (u1), [v0]"=&r" (v0), "=@cccc" (result), [u0]"=&r" (u0)
                                                    ^
  /Users/runner/work/Nim/Nim/pkgstemp/constantine/nimcache/libconstantine_static/@m..@sconstantine@smath@sarithmetic@sassembly@slimbs_asm_bigint_arm64.nim.c:117:51: error: invalid output constraint '=@cccc' in asm
  : [v1]"=&r" (v1), [u1]"=&r" (u1), [v0]"=&r" (v0), "=@cccc" (result), [u0]"=&r" (u0)
                                                    ^
  /Users/runner/work/Nim/Nim/pkgstemp/constantine/nimcache/libconstantine_static/@m..@sconstantine@smath@sarithmetic@sassembly@slimbs_asm_bigint_arm64.nim.c:193:51: error: invalid output constraint '=@cccs' in asm
  : [v1]"=&r" (v1), [u1]"=&r" (u1), [v0]"=&r" (v0), "=@cccs" (result), [u0]"=&r" (u0)
```

--- a/constantine/platforms/isa_arm64/macro_assembler_arm64.nim
+++ b/constantine/platforms/isa_arm64/macro_assembler_arm64.nim
@@ -373,13 +373,29 @@ func setOutputToFlag*(a: var Assembler_arm64, outputFlag: NimNode, carryOrBorrow
                   else: outputFlag
   let symStr = $nimSymbol
 
-  let desc = OperandDesc(
-    asmId: "",
-    nimSymbol: ident(symStr),
-    rm: carryOrBorrow,
-    constraint: asmOutputOverwrite)
-  desc.setConstraintDesc(nimSymbol)
-  a.operands.incl(desc)
+  if false:
+    # Unsupported before MacOS 15 (Apple Clang 16) and before Clang 17:
+    # https://github.com/mratsim/constantine/issues/516
+    # https://reviews.llvm.org/D149123
+    let desc = OperandDesc(
+      asmId: "",
+      nimSymbol: ident(symStr),
+      rm: carryOrBorrow,
+      constraint: asmOutputOverwrite)
+    desc.setConstraintDesc(nimSymbol)
+    a.operands.incl(desc)
+  else:
+    let cc = if carryOrBorrow == CarryFlag: ConditionCode.cs
+             else: ConditionCode.cc
+    let asmId = "[" & $nimSymbol & "]"
+    a.code &= "cset %" & asmId & ", " & $cc & '\n'
+    let desc = OperandDesc(
+      asmId: asmId,
+      nimSymbol: ident(symStr),
+      rm: Reg,
+      constraint: asmOutputOverwrite)
+    desc.setConstraintDesc(nimSymbol)
+    a.operands.incl(desc)
 
 func generate*(a: Assembler_arm64): NimNode =
   ## Generate the inline assembly code from
@@ -569,6 +585,15 @@ func codeFragment(a: var Assembler_arm64, instr: string, op0, op1: Operand, op2:
     a.operands.incl op0.desc
   if op1.desc.constraint != asmClobberedRegister:
     a.operands.incl op1.desc
+
+func codeFragment(a: var Assembler_arm64, instr: string, op: Operand, cc: ConditionCode) =
+  # Generate a code fragment
+  let off = a.getStrOffset(op)
+
+  a.code &= instr & " " & off & ", " & $cc & '\n'
+
+  if op.desc.constraint != asmClobberedRegister:
+    a.operands.incl op.desc
 
 func codeFragment(a: var Assembler_arm64, instr: string, op0, op1, op2: Operand, cc: ConditionCode) =
   # Generate a code fragment
@@ -812,6 +837,11 @@ func cmn*(a: var Assembler_arm64, lhs: Operand, rhs: Operand) =
   ## Compare-negative and set flags / condition code
   ## This uses ADDS and discards the result
   a.codeFragment("cmn", lhs, rhs)
+
+func cset*(a: var Assembler_arm64, dst: Operand, cc: ConditionCode) =
+  ## Store a condition code in a register
+  doAssert dst.isOutput(), $dst.repr
+  a.codeFragment("cset", dst, cc)
 
 func `and`*(a: var Assembler_arm64, dst: Operand, lhs: Operand, rhs: OperandReuse) =
   a.codeFragment("and", dst, lhs, rhs)

